### ISSUE TITLE
use Cloudflare

### ISSUE THREAD

**Issue by ctm** [ctm/mb2-doc#488]
Use Cloudflare to alleviate the (potential?) thundering herd problem.

I don't know how much of an issue it really is currently, but at some point it would be nice if the shareable urls could be used more just so I can get a feel for "real world" performance. However, if someone with a big following were to share a url, it could result in a bunch of people all downloading the big WASM file at once. Eventually such a thing should be delegated to a CDN, so it makes sense to investigate this now.

It turns out Cloudflare has a free plan and it's fairly easy to configure, so wound up doing this before even creating this issue.

**Comment by ctm** [comment]
Oops. This totally hosed gerdog and tegwin.  It may be that they're using ipv6. I switched the nameserver back to dnsimple but that doesn't appear to have expunged the v6. I'll look at this tomorrow, although the nl 2-7 bug needs to be looked at.

**Comment by ctm** [comment]
Looks like [disabling ipv6](https://support.cloudflare.com/hc/en-us/articles/229666767-Understanding-and-configuring-Cloudflare-s-IPv6-support) may be easy.

**Comment by ctm** [comment]
Nope. Turns out I'll have to do it [via their API](https://mediarealm.com.au/articles/cloudflare-disable-ipv6/).

**Comment by ctm** [comment]
I've gone ahead and disabled IPV6 in Cloudflare and changed devctm.com's nameservers to the Cloudflare nameservers (i.e., re-enabled Cloudflare). I'll send email to gerdog later in the day and see if he can run the demo w/o trouble.

**Comment by ctm** [comment]
Disabling IPV6 fixed the problem gerdog was having.

**Comment by ctm** [comment]
We had some weird issues around the first break. I suspect it's Cloudflare related. I think I need to rejigger things so that only our assets are served by Cloudflare and our WebSockets go directly to devctm.com. Logically, that's the way to do it even if there are no issues, since Cloudflare isn't helping with WebSockets in any way...

**Comment by ctm** [comment]
I just now turned off the proxying for devctm.com (and www.devctm.com which shouldn't be used), but haven't switched the nameservers away from Cloudflare.

To proxy just our assets, I think I have to jump through a few hoops and get a wildcard SSL certificate and then tell webpack to use a subdomain for just the assets. I don't configure Let's Encrypt or webpack often enough to be able to do this w/o looking up the details and futzing around, so even though both of those are fairly easy to do (I believe), they would still probably take me a couple hours and I don't know that should attempt them today.

**Comment by ctm** [comment]
Oh, and gerdog had reported (in the lobby):

> gerdog 🥃: cliff fyi: during the late tourney the table/clock reset happened again during the break just like the first tourney

So the good thing is the problematic behavior occurs often enough that it should be fairly easy to test the solution I propose above.

**Comment by ctm** [comment]
It may make more sense to simply have a separate domain for the websockets and then run everything else through Cloudflare.

I say this after having futzed around with alternate webpack configurations and then realizing that a bunch of the paths come from either our own static HTML or come from wasm. I'm sure that given enough time I could change our static HTML to use some JavaScript to hit assets.devctm.com when deployed to devctm.com and also figure out how to get wasm to do the right thing, but cleaving off websockets seems much easier.

**Comment by ctm** [comment]
Per my previous comment, I've reconfigured things so that `devctm.com` (and the now unused `assets.devctm.com`) is cached through Cloudflare, but `ws.devctm.com` is not. I've also used developer tools to verify that the websocket is indeed getting made through `ws.devctm.com`.

So, my guess is this will solve the problem we saw at the break nine days ago. I'm going to leave this ticket open until after this evening's game though, just to be sure.
